ui/src/components/admin/follow-up/UserGrid.js

```diff
@@ -0,0 +1,411 @@
+};
+
+const officialNameHeaderGetter = (params, parentName) => {
+  const headerName = 'Official Name';
+  if (params.location === 'csv') {
+    return `${parentName} ${headerName}`;
+  }
+  return headerName;
+};
+
+const membershipHeaderGetter = (params, columnName) =>
+  params.location === 'csv' ? `Membership ${columnName}` : columnName;
+
+const membershipOfficialNameGetter = (params) => {
+  if (
+    params &&
+    params.data &&
+    params.data.membershipInfo &&
+    params.data.membershipInfo[0]
+  ) {
+    const { officialName } = params.data.membershipInfo[0];
+    return officialName;
+  }
+
+  return '';
+};
+
+const membershipInfoGetter = (params) => {
+  if (params) {
+    const { colId } = params.colDef;
+    const data = params.data.membershipInfo[0]?.[colId];
+
+    if (Array.isArray(data)) return data[0] ?? '';
+
+    return data;
+  }
+};
+
+const membershipClassAttendanceValueGetter = (params) => {
+  if (
+    params?.data?.membershipInfo == null ||
+    !Array.isArray(params.data.membershipInfo) ||
+    params.data.membershipInfo[0] == null ||
+    !Array.isArray(params.data.membershipInfo[0]?.classAttendance)
+  )
+    return '';
+
+  return params.data.membershipInfo[0].classAttendance.join();
+};
+
+const membershipFilterGetter = (params) => {
+  if (params) {
+    const { colId } = params.colDef;
+    const dateStr = params.data.membershipInfo[0]?.[colId];
+    if (dateStr) {
+      return DateTime.fromFormat(dateStr, dateFromFormat).toJSDate();
+    }
+  }
+};
+
+// Baptism Getters
+const baptismPlaceGetter = (params) => {
+  if (
+    params &&
+    params.data &&
+    params.data.baptismInfo &&
+    params.data.baptismInfo[0]
+  ) {
+    const { baptismPlace } = params.data.baptismInfo[0];
+    return baptismPlace;
+  }
+
+  return '';
+};
+
+const baptismOfficialNameGetter = (params) => {
+  if (
+    params &&
+    params.data &&
+    params.data.baptismInfo &&
+    params.data.baptismInfo[0]
+  ) {
+    const { officialName } = params.data.baptismInfo[0];
+    return officialName;
+  }
+
+  return '';
+};
+
+const baptismInfoGetter = (params) => {
+  if (params) {
+    const { colId } = params.colDef;
+    return params.data.baptismInfo[0]?.[colId];
+  }
+};
+
+const baptismClassAttendanceValueGetter = (params) => {
+  if (
+    params?.data?.baptismInfo == null ||
+    !Array.isArray(params.data.baptismInfo) ||
+    params.data.baptismInfo[0] == null ||
+    !Array.isArray(params.data.baptismInfo[0]?.classAttendance)
+  )
+    return '';
+
+  return params.data.baptismInfo[0].classAttendance.join();
+};
+
+const baptismFilterGetter = (params) => {
+  if (params) {
+    const dateStr = params.data.baptismInfo[0]?.baptismDate;
+    if (dateStr) {
+      return DateTime.fromFormat(dateStr, dateFromFormat).toJSDate();
+    }
+  }
+};
+
+const tithelyIdGetter = (params) => {
+  const { tithelyId } = params.data;
+  if (!tithelyId || tithelyId === '') return '';
+
+  return tithelyId.join(',');
+};
+
+// Formatters
```

Comment:
actually all this formatters, getters, and setters is repeated from user table ya? should we have helpers file for functions and import it then? to avoid repeated code

Reply:
Yes. But that requires quite a bit of refactor.

Reply:
i guess can put //TODO for future then